homeassistant/components/schedule/__init__.py

```diff
@@ -0,0 +1,339 @@
+        if loaded_rules is None:
+            self.generate_defaults()
+            await self.async_save_rules()
+        else:
+            self.load_rules_dict(loaded_rules["rules"])
+            self.defaults = loaded_rules["defaults"]
+
+    def update_rule_listeners(self):
+        """Add listeners for all times that have rule changes."""
+        for unsub in self.unsubs:
+            unsub()
+
+        times_set = set()
+        for rule in self.rules:
+            times_set.add(rule.start)
+            times_set.add(rule.end)
+        self.unsubs = []
+        for time in times_set:
+            r_time = dt.parse_time(time)
+            self.unsubs.append(async_track_time_change(self.hass,
+                                                       self.exec_schedule,
+                                                       hour=r_time.hour,
+                                                       minute=r_time.minute,
+                                                       second=0))
+
+    async def async_save_rules(self):
+        """Save rules to storage."""
+        await self.store.async_save({
+            'rules': list(map(lambda r: r.to_dict(), self.rules)),
+            'defaults': self.defaults
+        })
+
+    def generate_defaults(self):
+        """Generate default values for the monitored entities."""
+        for entity in self.monitored:
+            domain = split_entity_id(entity)[0]
+            if domain == 'climate':
+                self.defaults[entity] = 20
+            elif domain in SCHEDULE_SWITCHABLE_DOMAINS:
+                self.defaults[entity] = False
+            else:
+                self.defaults[entity] = False
+
+    def exec_schedule(self, now):
+        """Execute the configured schedule."""
+        new_state = {}
+        # Aggregate active states from all rules
+        for rule in self.rules:
+            if rule.should_update(now):
+                if rule.entity in new_state:
+                    _LOGGER.warning(
+                        "Overlapping rules running for entity %s!",
+                        rule.entity)
+
+                else:
+                    new_state[rule.entity] = rule.value
+
+        for entity in self.monitored:
+            if entity in new_state:
+                value = new_state[entity]
+            else:
+                value = self.defaults[entity]
+            self.setValue(entity, value)
+        _LOGGER.info("Schedule executed, result = %s", json.dumps(new_state))
+        self.active_status = new_state
+
+    def setValue(self, entity, value):
+        """Set the entity value.
+
+        Calls a service based on the entity's domain to set
+        the value
+        """
+        domain = domain = split_entity_id(entity)[0]
+        if domain == "climate":
+            self.hass.loop.create_task(
+                self.hass.services.async_call(
+                    'climate',
+                    'set_temperature',
```

Comment:
How will this work if you say want a schedule to set your heating to heat using this?

Setting the temperature only works if you have a system that is constantly set to heat or cool etc.

_Apologies for jumping in btw_ 😃